### PR TITLE
Add CarsOwnedFilter and CarsPreferredFilter

### PR DESCRIPTION
**Pull request by OhJunMing**
Shared parent CarFilter class to enclose common methods for matching carBrand, carType attributes

### CODE REVIEW

File: src/main/java/seedu/address/logic/filters/CarFilter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Parses filter command for c/, cp/ prefixes.
+     * @param car
+     * @return
+     */
+    public static Car parseCar(String car) {
+        String trimmedCar = car.trim();
+        String[] carDetails = trimmedCar.split("\\+");
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/logic/filters/CarFilter.java:27]: Does "+Subaru" produce the array ["","Subaru"]

Comment (rajobasu) [reply src/main/java/seedu/address/logic/filters/CarFilter.java:27]: ig so, we can add this as a testcase handling later on no worries

File: src/main/java/seedu/address/logic/filters/CarsOwnedFilter.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     * @param customer
+     * @return
+     */
+    @Override
+    public boolean test(Customer customer) {
+        requireNonNull(customer);
+        return customer.getCarsOwned()
+                .keySet()
+                .parallelStream()
+                .anyMatch(x -> carFilterPredicate(x, parseCar(filterString)));
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/logic/filters/CarsOwnedFilter.java:33]: Could replace `x` with `customerCar`, but its not exactly block worthy this close to deadline. Fix soon during 1.4 please.

**Review by rajobasu — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/filters/CarsPreferredFilter.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+     *
+     * @param customer
+     * @return
+     */
+    @Override
+    public boolean test(Customer customer) {
+        requireNonNull(customer);
+        return customer.getCarsPreferred()
+                .parallelStream()
+                .anyMatch(x -> carFilterPredicate(x, parseCar(filterString)));
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/logic/filters/CarsPreferredFilter.java:32]: Could replace `x` with `customerCar`, like in the other filter.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -193,16 +193,15 @@ public static Tag parseTag(String tag) throws ParseException {
     public static Car parseCarPreferred(String car) throws ParseException {
         String trimmedCar = car.trim();
         String[] carDetails = trimmedCar.split("\\+");
+        if (carDetails.length != 2) {
+            throw new ParseException(Car.MESSAGE_CONSTRAINTS);
+        }
```
Comment (nighoggDatatype) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:198]: Good catch

**Review by nighoggDatatype — COMMENTED** [review]
Just some changes you probably want to make in 1.4

**Pull request merged**